### PR TITLE
feat: adding base setters and tests for minisales

### PR DESCRIPTION
**Pull request by Nilay27**
* adding methods for `setPurchaseHook()` and `setPrice()` along with respective getters and tests.

### CODE REVIEW

File: apps/smart-contracts/token/contracts/mini-sales/MiniSales.sol [hunk 1]
```diff
@@ -8,6 +8,8 @@ import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.
 contract MiniSales is IMiniSales, SafeOwnableUpgradeable, ReentrancyGuardUpgradeable {
   IERC20 private _saleToken;
   IERC20 private _paymentToken;
+  IPurchaseHook _purchaseHook;
+  uint256 _price;
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/contracts/mini-sales/MiniSales.sol:12]: nit: keep order consistent
```suggestion
  uint256 _price;
  IPurchaseHook _purchaseHook;
```

File: apps/smart-contracts/token/contracts/mini-sales/MiniSales.sol [hunk 2]
```diff
@@ -20,9 +22,13 @@ contract MiniSales is IMiniSales, SafeOwnableUpgradeable, ReentrancyGuardUpgrade
...
 
-  function setPrice(uint256 _newPrice) external override onlyOwner {}
+  function setPrice(uint256 _newPrice) external override onlyOwner {
+    _price = _newPrice;
+  }
 
-  function setPurchaseHook(IPurchaseHook _newPurchaseHook) external override onlyOwner {}
+  function setPurchaseHook(IPurchaseHook _newPurchaseHook) external override onlyOwner {
+    _purchaseHook = _newPurchaseHook;
+  }
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/contracts/mini-sales/MiniSales.sol:31]: events missing - but you can leave a TODO and add in a followup

File: apps/smart-contracts/token/test/MiniSales.test.ts [hunk 3]
```diff
@@ -59,5 +64,100 @@ describe('=> MiniSales', () => {
...
       expect(await miniSales.getPaymentToken()).to.eq(paymentToken.address)
     })
+
+    it('sets purchase hook to zero address', async () => {
+      expect(await miniSales.getPurchaseHook()).to.eq(ZERO_ADDRESS)
+    })
+
+    it('sets price to zero', async () => {
+      expect(await miniSales.getPrice()).to.eq(0)
+    })
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/test/MiniSales.test.ts:74]: nit:
```suggestion
    it('sets price to zero', async () => {
      expect(await miniSales.getPrice()).to.eq(0)
    })

    it('sets purchase hook to zero address', async () => {
      expect(await miniSales.getPurchaseHook()).to.eq(ZERO_ADDRESS)
    })
```

File: apps/smart-contracts/token/test/MiniSales.test.ts [hunk 4]
```diff
@@ -59,5 +64,100 @@ describe('=> MiniSales', () => {
...
+
+      expect(await miniSales.getPurchaseHook()).to.eq(JUNK_ADDRESS)
+
+      await miniSales.connect(owner).setPurchaseHook(JUNK_ADDRESS)
+
+      expect(await miniSales.getPurchaseHook()).to.eq(JUNK_ADDRESS)
+    })
+  })
+
+  describe('# setPrice', () => {
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/test/MiniSales.test.ts:121]: nit: move above hook tests

**Review by XavierEkkel — APPROVED** [review]
LGTM after comments

File: apps/smart-contracts/token/contracts/mini-sales/MiniSales.sol (hunk 2 above)
Comment (ramenforbreakfast) [reply apps/smart-contracts/token/contracts/mini-sales/MiniSales.sol:36]: I will submit this in a follow up PR.

File: apps/smart-contracts/token/contracts/mini-sales/MiniSales.sol [hunk 5]
```diff
@@ -8,6 +8,8 @@ import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.
 contract MiniSales is IMiniSales, SafeOwnableUpgradeable, ReentrancyGuardUpgradeable {
   IERC20 private immutable _saleToken;
   IERC20 private immutable _paymentToken;
+  uint256 _price;
+  IPurchaseHook _purchaseHook;
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/token/contracts/mini-sales/MiniSales.sol:12]: These need to be private, will add this as well.

**Review by ramenforbreakfast — APPROVED** [review]

**Pull request merged**